index.html

```diff
@@ -126,7 +127,35 @@
+                                		<li>Alternatywą dla Dependency Injection są <a href="http://laravel.com/docs/5.1/facades">Fasady</a>. Ich nadużywanie prowadzi do sytuacji, w której trudno jest określić zależności danej klasy. Planowane jest wprowadzenie funkcjonalności pozwalającej na <a href="http://laravel.com/docs/master/blade#service-injection">dostęp do serwisów z poziomu template'ów.</a></li>
+                                        <li>Ze względu na Fasady w <a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Support/Facades/Facade.php#L42">produkcyjnym kodzie</a> znajdują się metody służące wyłącznie testowaniu.</li>
+                                        <li>Zapisywanie obiektu do bazy realizowane jest za pomocą <a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Database/Eloquent/Model.php#L1185">statycznej metody save()</a>, której mockowanie w testach jednostkowych jest mocno utrudnione.</li>
+                                        <li>Przykładem nadużywanie statycznego dostępu jest klasa odpowiedzialna za <a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Support/Str.php">operacje na ciągach znaków</a>.</li>
+                                </ol>
+                                <p>Metody magiczne:</p>
+                                <ol>
+                                        <li><a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Database/Eloquent/Builder.php">Eloquent\Builder</a> nie dziedziczy po <a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Database/Query/Builder.php">Query\Builder</a>. W zamian  metoda <a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Database/Eloquent/Builder.php#L920">__call()</a> powoduje wywoływanie metod nieistniejących w Eloquent\Builder na <a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Database/Eloquent/Builder.php#L17">właściwości</a> tej klasy będącej instancją Query\Builder.</li>
+                                        <li>W podobny sposób działa <a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Auth/AuthManager.php">AuthManager</a>, który pozwala niejawnie <a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Support/Manager.php#L137">wykonywać</a> dzialanie na <a href="http://laravel.com/docs/5.0/authentication#retrieving-the-authenticated-user">klasie</a>, której instancja jest elementem <a href="https://github.com/laravel/framework/blob/v5.0.32/src/Illuminate/Support/Manager.php#L27">tablicy</a> będącej jego właściwością. <a href="">Dyskusja</a></li>
+                                        <li>Opisane powyżej podejście znacznie utrudnia debugowanie i zrozumienie wewnętrznych mechanizmów frameworka.</li>
+                                </ol>
```

Comment:
https://github.com/laravel/framework/blob/5.1/src/Illuminate/Support/helpers.php - to też bym gdzieś wrzucił